application/hubspot/v0/config/tasks.json

```diff
@@ -0,0 +1,1149 @@
+    "instillShortDescription": "Retrieve all the messages inside a thread (conversation inbox)",
+    "input": {
+      "description": "Input thread ID",
+      "instillUIOrder": 0,
+      "properties": {
+        "thread-id": {
+          "description": "Input thread ID",
+          "instillAcceptFormats": ["string"],
+          "instillUIMultiline": true,
+          "instillUIOrder": 0,
+          "instillUpstreamTypes": ["value", "reference", "template"],
+          "title": "Thread ID",
+          "type": "string"
+        }
+      },
+      "required": ["thread-id"],
+      "title": "Input",
+      "type": "object"
+    },
+    "output": {
+      "description": "All messages in the thread",
+      "instillUIOrder": 0,
+      "properties": {
+        "results": {
+          "description": "An array of messages",
+          "instillUIOrder": 0,
+          "title": "Messages",
+          "type": "array",
+          "items": {
+            "title": "Messages Details",
+            "type": "object",
+            "properties": {
+              "created-at": {
+                "description": "message to start a conversation",
+                "instillFormat": "string",
+                "instillUIOrder": 0,
+                "title": "Start Conversation Message",
+                "type": "string"
+              },
+              "senders": {
+                "description": "Senders",
+                "instillFormat": "array",
+                "instillUIOrder": 1,
+                "title": "Senders",
+                "type": "array",
+                "items": {
+                  "title": "Sender's information",
+                  "$ref": "#/$defs/thread/thread-user-information"
+                }
+              },
+              "recipients": {
+                "description": "Recipients",
+                "instillFormat": "array",
+                "instillUIOrder": 2,
+                "title": "Recipients",
+                "type": "array",
+                "items": {
+                  "title": "Recipient's information",
+                  "$ref": "#/$defs/thread/thread-user-information"
+                }
+              },
+              "text": {
+                "description": "when a conversation starts",
+                "instillFormat": "string",
+                "instillUIOrder": 3,
+                "required": [],
+                "title": "Start Date",
+                "type": "string"
+              },
+              "subject": {
+                "description": "Date of the last message",
+                "instillFormat": "string",
+                "instillUIOrder": 4,
+                "required": [],
+                "title": "Last Date",
+                "type": "string"
+              }
+            },
+            "required": ["created-at", "text", "subject"]
+          }
+        }
+      },
+      "required": ["results"],
+      "title": "Output",
+      "type": "object"
+    }
+  },
+  "TASK_RETRIEVE_ASSOCIATION": {
```

Comment:
It is a good design for some specific use cases!
I believe some users will benefit by this TASK.
Cool!